### ISSUE TITLE
size_average in cross_entropy2d

### ISSUE THREAD

**Issue by maoxin** [meetshah1995/pytorch-semseg#82]
In the code of cross_entropy2d, if `size_average == True`, then `loss /= mask.data.sum()`. The weight, however, is ignored. I think it should be changed to `loss /= self.weight[target.data].sum()` if there is a weight provided. Thanks!

**Comment by oujieww** [comment]
hi, i also have problem in loss function, For 'Expected object of type torch.cuda.FloatTensor but found type torch.cuda.LongTensor for argument #3 'other''
so i use 'loss /= mask.data.sum().float()'  ,but i want to know how do u use 'loss /= self.weight[target.data].sum()'     for the full code of your loss funciton. can u give me ? and how about the result?
